### DISCUSSION TITLE
What is libGLESv2.dll needed for?

### DISCUSSION THREAD

**Discussion by Xan-Kun** [cefsharp/CefSharp#4811] [Q&A] [answered]
Hello and first of all thanks for these amazing libraries!
My question is:
What exactly is `libGLESv2.dll` needed for?

I have a `WinForm` project using `CefSharp.WinForms.NETCore` (124.3.80).
The target is `win-x64` with `net8.0-windows7.0`.

CefSharp copies the `libGLESv2.dll` to the build dir and if it's there, I get `AccessViolationException`s from a different package.
What confuses me: If I just delete the dll in the build directory, everything seems to work, including the various browser instances.
And I get no more `AccessViolationException`s.

So I am just curious how and why this file is there.

Thanks so much for your time, much appreciated,
and please excuse if the answer is super obvious and I just managed to not find it :-)

**Comment by Xan-Kun** [comment, marked as answer]
Just realized that HW acceleration is gone, when deleting the file.
As to be expected 😉 
Sry.

**Reply by Xan-Kun** [reply]
Still doesnt solve the problem with the exceptions, but I am a step further.

**Comment by amaitland** [comment]
See https://github.com/cefsharp/CefSharp/wiki/Output-files-description-table-%28Redistribution%29 for a list of dependencies and a description of their purpose.

> ANGLE support
> Support for rendering of HTML5 content like 2D canvas, 3D CSS and WebGL. Without these files the aforementioned capabilities may fail.

It's used for hardware accelerated rendering.

Technically `libGLESv2.dll` is optional. Try `--use-gl=angle --use-angle=swiftshader` to force using `SwANGLE`

> I get AccessViolationExceptions from a different package.

Does the exception only occur when you are using this different package?

How does the https://github.com/cefsharp/CefSharp.MinimalExample behave?

**Reply by Xan-Kun** [reply]
Thanks for the hint. I found it. But the problem with persists. I will open a new issue.
